### PR TITLE
Always setup rpath

### PR DESCRIPTION
**Pull request by Neumann-A**
Reasoning:
a) vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY) -> builds .so
b) static libs links the above
c) a code generator links the above (e.g. qsb in #27279 complaining about not finding libGLX.so.0 while building qtdeclarative)

as such vcpkg needs to always fix the rpath because otherwise required build tools might be looking for shared libraries. 

cc @Osyotr

### CODE REVIEW

**Comment by Osyotr** [comment]
> vcpkg needs to always fix the rpath 

This was discussed in the original PR. https://github.com/microsoft/vcpkg/pull/23035#discussion_r921583704
I would like to make it default on `x64-linux`, too.
@vicroms

**Comment by Neumann-A** [comment]
hmm @autoantwort does vcpkg need a similar elf patch on osx? Seems like the current approach of @Osyotr doesn't work there?

**Comment by autoantwort** [comment]
I have no idea. I don't use `arm64-osx-dynamic`.

**Comment by Neumann-A** [comment]
> I have no idea. I don't use arm64-osx-dynamic.

This is not about `osx-dynamic` but more about `vcpkg_check_linkage(ONLY_DYNAMIC_LIBRARY)` and if we have to do the same as linux does with fixing rpaths on osx. 

https://stackoverflow.com/questions/12521802/print-rpath-of-an-executable-on-macos
https://github.com/bazelbuild/bazel/pull/4589
https://wincent.com/wiki/@executable_path,_@load_path_and_@rpath

File: scripts/ports.cmake [hunk 1]
```diff
@@ -146,6 +146,9 @@ if(CMD STREQUAL "BUILD")
 
     include("${CURRENT_PORT_DIR}/portfile.cmake")
     if(DEFINED PORT)
+        if(NOT DEFINED VCPKG_FIXUP_ELF_RPATH AND NOT VCPKG_TARGET_IS_WINDOWS)
```
Comment (Neumann-A) [inline scripts/ports.cmake:149]:
```suggestion
        if(NOT DEFINED VCPKG_FIXUP_ELF_RPATH AND VCPKG_TARGET_IS_LINUX)
```

File: scripts/cmake/z_vcpkg_fixup_rpath.cmake [hunk 2]
```diff
@@ -33,6 +33,11 @@ function(z_vcpkg_fixup_rpath_in_dir)
                 set(current_prefix "${CURRENT_PACKAGES_DIR}/debug")
             endif()
 
+            # TODO: Read out current rpath
+            #       - nuke out any vcpkg related paths
+            #       - remove just $ORIGIN paths (because we add them any way)
+            #       - prepend our paths
```
Comment (Neumann-A) [inline scripts/cmake/z_vcpkg_fixup_rpath.cmake:39]: @Osyotr What is your opinion on this TODO?

Comment (Osyotr) [reply scripts/cmake/z_vcpkg_fixup_rpath.cmake:39]: Do you expect something other than `$ORIGIN` and `/path/to/vcpkg_installed` to be in the path?
If it's an absolute path, I don't think we should keep it. On the other hand, `$ORIGIN` stuff can theoretically point to something other than `lib/`, so we could try to keep it.

**Comment by Adela0814** [comment]
Pinging @Neumann-A for response. Is work still being done for this PR?

**Comment by Neumann-A** [comment]
@Adela0814. Ideally the TODO needs to be taken care of.

File: scripts/cmake/z_vcpkg_fixup_rpath.cmake (hunk 2 above)
Comment (Neumann-A) [reply scripts/cmake/z_vcpkg_fixup_rpath.cmake:39]: > Do you expect something other than $ORIGIN and /path/to/vcpkg_installed to be in the path?

Yes if the user set it up in the toolchain itself.

Comment (Osyotr) [reply scripts/cmake/z_vcpkg_fixup_rpath.cmake:39]: I don't think we will be able to tell whether a path was set in toolchain or in library itself.

Comment (Neumann-A) [reply scripts/cmake/z_vcpkg_fixup_rpath.cmake:39]: in #36056 I came to the conclusion that it is enough to support rpaths which point to the installed tree itself.

**Comment by Neumann-A** [comment]
superseded by https://github.com/microsoft/vcpkg/pull/36056

**Pull request closed without merging**